AM62ax ThreadX. Add Iperf and TSN support. General performance improvements to driver integration for NETXDUO.

jblanchard-jblopen opened the pull request:
This PR adds IPERF and multi host TSN support to NETXDUO as well as performance improvements and various fix to the NETXDUO driver integration for the AM62AX CPSW controller. The PR also adds examples for the new features as well as documentation for those examples.

Note that this PR enables building the TSN library for the A53 on the AM62ax which is disabled by default.

shiva-ti commented on source/kernel/freertos/FreeRTOS-Plus-CLI/FreeRTOS_CLI.c:
@jblanchard-jblopen why are these FreeRTOS CLI files included ?

jblanchard-jblopen replied:
Indeed those files were staged by mistake. PR update to remove them.

jblanchard-jblopen committed:
am62ax: threadx: TSN support, IPERF and Performance improvements and fixes to the netxduo integration.

- implement zero-copy rx/tx.
- fix enet packet descriptor added to the wrong queue when netx packet is unavailable.
- enable https
- enable TLS1.3
- enable self-signed certificates
- set packet header alignment to cache alignment.
- Add IPERF module.
- Add IPERF, GPTP and HTTPS examples.
- Change default Threfx examples tack priority.
- Fix NETXDUO kernel timebase period.
- Documented added examples.
- Enabled TSN build for A53.

Fixes: SITSW-5614
Signed-off-by: Edouard Haas <edouard.haas@jblopen.com>
